### PR TITLE
Holmanb/integration test fix ppa

### PR DESCRIPTION
**Pull request by holmanb**
```
Update apt test for new ppa url in jammy
```

## Additional Context
Looks like [we're not the only ones](https://www.mail-archive.com/ubuntu-bugs@lists.ubuntu.com/msg5999672.html) affected by this behavior change. 

Test failures look like this:
```
07:51:15 ___________________________ TestApt.test_ppa_source ____________________________
07:51:15 
07:51:15 self = <test_apt.TestApt object at 0x7f7b6878fda0>
07:51:15 class_client = <tests.integration_tests.instances.IntegrationInstance object at 0x7f7b6554c940>
07:51:15 
07:51:15     def test_ppa_source(self, class_client: IntegrationInstance):
07:51:15         """Test the apt ppa functionality.
07:51:15     
07:51:15         Ported from
07:51:15         tests/cloud_tests/testcases/modules/apt_configure_sources_ppa.py
07:51:15         """
07:51:15         release = ImageSpecification.from_os_image().release
07:51:15         ppa_path_contents = class_client.read_from_file(
07:51:15             "/etc/apt/sources.list.d/"
07:51:15             "simplestreams-dev-ubuntu-trunk-{}.list".format(release)
07:51:15         )
07:51:15     
07:51:15 >       assert (
07:51:15             "http://ppa.launchpad.net/simplestreams-dev/trunk/ubuntu"
07:51:15             in ppa_path_contents
07:51:15         )
07:51:15 E       AssertionError: assert 'http://ppa.launchpad.net/simplestreams-dev/trunk/ubuntu' in 'deb https://ppa.launchpadcontent.net/simplestreams-dev/trunk/ubuntu/ jammy main\n# deb-src https://ppa.launchpadcontent.net/simplestreams-dev/trunk/ubuntu/ jammy main'
07:51:15 
```

### CODE REVIEW

File: tests/integration_tests/modules/test_apt.py [hunk 1]
```diff
@@ -156,6 +156,8 @@ def test_ppa_source(self, class_client: IntegrationInstance):
         assert (
             "http://ppa.launchpad.net/simplestreams-dev/trunk/ubuntu"
             in ppa_path_contents
+            or "https://ppa.launchpadcontent.net/simplestreams-dev/trunk/ubuntu/"
```
Comment (blackboxsw) [inline tests/integration_tests/modules/test_apt.py:159]: Looks like flake8 is unhappy due to line length. Maybe something that also is more specific to Jammy-only as far as conditionals?

That way our test is more prescriptive/strict on each series so we know exactly what
we expect and where we expect it.

```diff
diff --git a/tests/integration_tests/modules/test_apt.py b/tests/integration_tests/modules/test_apt.py
index 6b3a8b7c..46a5e301 100644
--- a/tests/integration_tests/modules/test_apt.py
+++ b/tests/integration_tests/modules/test_apt.py
@@ -153,8 +153,9 @@ class TestApt:
             "simplestreams-dev-ubuntu-trunk-{}.list".format(release)
         )
 
+        host = "launchpad" if release == "jammy" else "launchpadcontent"
         assert (
-            "http://ppa.launchpad.net/simplestreams-dev/trunk/ubuntu"
+            f"http://ppa.{host}.net/simplestreams-dev/trunk/ubuntu"
             in ppa_path_contents
         )

```

**Review by blackboxsw — CHANGES_REQUESTED** [review]
minor thought on avoiding the linelength and making the test more strict and jammy-only.

Your call, but we probably need to sort formating/flake errors somehow.

File: tests/integration_tests/modules/test_apt.py (hunk 1 above)
Comment (holmanb) [reply tests/integration_tests/modules/test_apt.py:159]: Oops, I fixed format locally but forgot to push the commit. I'll include your suggestion in my next push.

**Review by blackboxsw — APPROVED** [review]
LGTM

File: tests/integration_tests/modules/test_apt.py [hunk 2]
```diff
@@ -152,9 +152,9 @@ def test_ppa_source(self, class_client: IntegrationInstance):
             "/etc/apt/sources.list.d/"
             "simplestreams-dev-ubuntu-trunk-{}.list".format(release)
         )
-
+        host = "launchpad" if release == "jammy" else "launchpadcontent"
```
[the remark this answers is not in the document]
Comment (holmanb) [reply tests/integration_tests/modules/test_apt.py:155]: Looks like you're right. Fix PR here: https://github.com/canonical/cloud-init/pull/1296

**Pull request merged**